### PR TITLE
Calculate version using setuptools scm

### PR DESCRIPTION
**Pull request by Hook25**
## Description

Since bumpversion modifies the version history of the repository as part of setting version metadata in place, and since our CI needs are made complex by lab automation being done by Jenkins jobs that can be executed once there is a package available that are used to validate the state of the package for promoting it to the next stability level (edge → beta → stable kind of flow), we are best off migrating away from bumpversion to a tool that does not modify the state of source code in the repository but can operate with tags

## Resolved issues

Resolves: [CHECKBOX-613](https://warthogs.atlassian.net/browse/CHECKBOX-613)
Fixes: https://github.com/canonical/checkbox/issues/524

## Documentation

N/A

## Tests

This was tested offline via metabox both in legacy an remote runs on jammy and kinetic

[CHECKBOX-613]: https://warthogs.atlassian.net/browse/CHECKBOX-613?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Comment by yphus** [comment]
> @yphus (or whoever is reviewing this):
> 
> I'm not super sure about these two changes especially, so please give a look if they make sense: in: `checkbox-support/tox.ini` I had to add `requests-unixsocket == 0.3.0` as a tox dependency because I tried for hours to debug why it is not installed by the previous install comand but to no success. If you see what is wrong with it, we may be able to remove this dep from this file
> 
If tox is happy with `requests-unixsocket == 0.3.0` for [testenv:py35] let's adopt your fix.

> in: `docs/.sphinx/requirements.txt` and `docs/Makefile` I had to do a weird hack to have the correct version installed of `checkbox-ng` while building the docs, so that it can be correctly recovered. The idea is that readthedocs launches the install requirements command from the repo root, so in order to install the module I have to put a relative requirement but the makefile was not in the same dir as the website, so I had to change it

I'm not sure we need to import `checkbox_ng` now that the version will be retrieve using tags and setuptools-scm. I thought this [snippet](
https://github.com/pypa/setuptools_scm/#programmatic-usage) would satisfy sphinx `conf.py`.

One last thing, with [workflows](https://github.com/canonical/checkbox/blob/main/.github/workflows/automatic-check.yml) inherited from the starter pack, I think you can drop sphinx and doc build tests from tox configs.

**Comment by Hook25** [comment]
I intentionally didn't use it programmatically due to the following: https://github.com/pypa/setuptools_scm/#usage-from-sphinx

**Review by mz2 — CHANGES_REQUESTED** [review]
Please cover the way versioning behaves, in words beside the implementation (README? CONTRIBUTING.md? is therea closer place for covering release process methodology?)

File: checkbox-ng/pyproject.toml
Comment (mz2) [inline checkbox-ng/pyproject.toml:8]: Please enter a name and email for someone who if actually emailed doesn't bounce.

**Comment by pieqq** [comment]
> is therea closer place for covering release process methodology?

The [Release process README file](https://github.com/canonical/checkbox/blob/main/tools/release/README.md) is probably what you're looking for.

**Review by yphus — APPROVED** [review]

**Review by yphus — APPROVED** [review]
+1

**Review by yphus — APPROVED** [review]
+1 (re-approving after merging PR #578 )

File: checkbox-core-snap/README.md
Comment (mz2) [inline checkbox-core-snap/README.md:6]:
```suggestion
This section covers how to build the snap
```

Comment (mz2) [reply checkbox-core-snap/README.md:6]: (correct would be `setting`, not `settings`, but it is also redundant)

File: checkbox-core-snap/README.md
Comment (mz2) [inline checkbox-core-snap/README.md:18]: Perhaps explain what `prepare.sh` is about. For example:

"After installing all the dependencies and cloning the repo, to output the snapcraft.yaml for `core22` as an example, do the following:"

File: checkbox-core-snap/README.md
Comment (mz2) [inline checkbox-core-snap/README.md:35]:
```suggestion
To test the build one must install it and see if the content is correct.
```

File: checkbox-core-snap/README.md
Comment (mz2) [inline checkbox-core-snap/README.md:36]:
```suggestion
One way to do it is the following:
```

Comment (mz2) [reply checkbox-core-snap/README.md:36]: (cruft)

File: checkbox-core-snap/README.md
Comment (mz2) [inline checkbox-core-snap/README.md:41]:
```suggestion
Now we have installed the core snap, but we cannot use it directly. To use it
```

File: checkbox-core-snap/README.md
Comment (mz2) [reply checkbox-core-snap/README.md:53]: (Typo + the term "guide" weirds me out a bit since I might call the whole set of of documentation a "guide")

File: .github/workflows/deb-daily-builds.yml
Comment (mz2) [inline .github/workflows/deb-daily-builds.yml:17]: Is this fork of `ppa-dev-tools` going to be long lived?

File: checkbox-core-snap/prepare.sh
Comment (mz2) [inline checkbox-core-snap/prepare.sh:51]: Out of curiosity, what's the grepping achieving?

File: checkbox-core-snap/series16/snap/snapcraft.yaml
Comment (mz2) [inline checkbox-core-snap/series16/snap/snapcraft.yaml:64]:
```suggestion
      # The version string is written to a file by prepare.sh as a pre-build step
      # since it is computed based on git repo metadata not available for a snapcraft build
```

**Review by mz2 — APPROVED** [review]
There are some typos and awkward phrases which I suggested changes to in docs. A bunch of other nitpicks. Please apply as you see fit and merge.

File: checkbox-core-snap/prepare.sh
Comment (Hook25) [reply checkbox-core-snap/prepare.sh:51]: on old versions of setuptools_scm instead of just printing the calculated version the output is something like `Caluclated version vX.XX...`, with grep just takes the last part of that

File: .github/workflows/deb-daily-builds.yml
Comment (Hook25) [reply .github/workflows/deb-daily-builds.yml:17]: Unrelated to this PR, but we could actually go back to the upstream version now that every change I made is in main

File: checkbox-core-snap/README.md
Comment (Hook25) [reply checkbox-core-snap/README.md:42]: I can change this for front-end snap if it is clearer, what I want to say it that we need a snap that uses the core

Comment (mz2) [reply checkbox-core-snap/README.md:42]: Frontend makes sense indeed. If there's a place for this in the user docs, do add that too.

Comment (Hook25) [reply checkbox-core-snap/README.md:48]: Updated here, given that this whole readme should be ported to the docs as well I think that the operation of documenting the whole frontend / core business should be done there. Also, we should consider renaming core to something like runtime, but that is unrelated to this PR imo

**Pull request merged**
